### ISSUE TITLE
Database Permission Issue - was working now isnt

### ISSUE THREAD

**Issue by acoale** [firebase/quickstart-unity#464]
## Please fill in the following fields:
Unity editor version:  2018.4.8f1
Firebase Unity SDK version: 5.4.4
Firebase plugins in use (Auth, Database, etc.): Auth, database, storage, analytics
Additional SDKs you are using (Facebook, AdMob, etc.): Mapbox
Platform you are using the Unity editor on (Mac, Windows, or Linux): Windows
Platform you are targeting (iOS, Android, and/or desktop): Android

## Please describe the issue here:
System.AggregateException: One or more errors occurred. ---> Firebase.Database.DatabaseException: Firebase Database error: Permission denied
   --- End of inner exception stack trace ---
---> (Inner Exception #0) Firebase.Database.DatabaseException: Firebase Database error: Permission denied<---

UnityEngine.Debug:Log(Object)
LoginPanel:<PullDatabase>b__34_0(Task`1) (at Assets/Scripts/Panels/LoginPanel.cs:130)
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback()

## Please answer the following, if applicable:
N/A

The issue is that I changed from .net35 to .net4x I updated the firebase plugin to reflect the appropriate .net version. I use auth to sign in first then access the database. Its textbook and pulled almost completely from docs. I do get a user signed in successfully log from AUTH. 

PERMISSIONS:
{
  "rules": {
    ".read": "auth != null",
    ".write": "auth != null"
  }
} 

Where the permission is denied:

```
private void GetUserProfile()
    {
        auth = FirebaseAuth.DefaultInstance;
        auth.SignInWithEmailAndPasswordAsync(email.text, password.text).ContinueWith(task =>
        {
            if (task.IsCanceled)
            {
                Debug.LogError("SignInWithEmailAndPasswordAsync was canceled.");
                ErrorMessage.Instance.ErrorHandler("Error: Sign in was canceled.");
                //Error.ErrorHandler("Sign In Canceled.");
                return;
            }
            if (task.IsFaulted)
            {
                Debug.LogError("Sign In With Email And PasswordAsync encountered an error: " + task.Exception);
                ErrorMessage.Instance.ErrorHandler("Error: Email and/or Password do not match.");
                Debug.Log("Test");
                return;
            }
            FirebaseUser newUser = task.Result;
            Debug.LogFormat("User signed in successfully: {0} ({1})",
                "User Display Name: " + newUser.DisplayName, "User ID: " + newUser.UserId);
            userID = newUser.UserId;
            Debug.Log(userID);
            PullDatabase();
            email.text = "";
            password.text = "";
        });
    }

private void PullDatabase()
    {
        Debug.Log("Pulling Database:...");
        FirebaseDatabase.DefaultInstance.GetReference("Users").Child(userID)
        .GetValueAsync().ContinueWith(task =>
        {
            if (task.IsFaulted)
            {
                Debug.Log("Could Not Sign in " + userID);
                Debug.Log(task.Exception);
            }
            else if (task.IsCompleted)
            {
                DataSnapshot snapshot = task.Result;                
            }
        });
    }
```

**Comment by acoale** [comment]
So an update to this. I created a new project in firebase to test permissions and linked the current project to that instead. Long story short, I have two methods that deal with registering users and/or pulling user details from the database. I was able to register a user and write their information to the database. BUT once I restarted the app and tried to pull the database (by logging in) I get the permission denied error.

**Comment by alexames** [comment]
Hi acoale,

Is there a reason you're using 5.4.4 instead of a newer release of the SDK? According to our [release notes](https://firebase.google.com/support/release-notes/unity) there have been a handful of bugfixes to authenticating in database. It's likely the issue you're seeing could be solved by just upgrading to the newest release. Nothing you've posted here immediately jumps out as being obviously wrong, so you might just be running into a bug in the version of the SDK you're using.

**Comment by acoale** [comment]
When I was trying to update I got the .dll error. I was updating to 6.4, I am currently on 6.5 as that version seemed to have fixed that issue, I am running on 2018.4.8f1 but only running .Net 3.x as I have an issue where Unity is not running code that is working on 3.x and I am trying to track down the reason why.
